src/jobs/interceptors/create-job-v3-mapping.interceptor.ts

```diff
@@ -24,55 +28,116 @@ interface JobParams {
+      }
+      const jobParams: JobParams = {
+        datasetList: dtoV3.datasetList ?? [],
+        ...dtoV3.jobParams,
       };
-    }
-    // ensure compatibility with the FE, which provides the username field in jobParams
-    // and compatibility the v4 which requires ownerUser and ownerGroup for jobs created by user
-    if (Object.keys(jobParams).includes("username")) {
-      const jobUser = await this.usersService.findByUsername2JWTUser(
-        jobParams.username as string,
-      );
-      newBody = {
-        ...newBody,
-        ownerUser: jobUser?.username,
-        ownerGroup: jobUser?.currentGroups[jobUser.currentGroups.length - 1],
+      // to preserve the executionTime field, if provided, add it to jobParams
+      if (dtoV3.executionTime) {
+        jobParams.executionTime = dtoV3.executionTime;
+      }
+      // assign jobParams and contactEmail to the new body
+      let newBody: CreateJobDto = {
+        type: dtoV3.type,
+        jobParams: jobParams,
       };
+      if (dtoV3.emailJobInitiator || requestUser) {
+        newBody = {
+          ...newBody,
+          contactEmail: dtoV3.emailJobInitiator ?? requestUser.email,
+        };
+      }
+      // ensure compatibility with the FE, which provides the username field in jobParams
+      // and compatibility with v4 which requires ownerUser in the dto of jobs created by normal users
+      // if username is not provided, use the username from the request user
+      let jobUser: JWTUser | null = null;
+      if (Object.keys(jobParams).includes("username")) {
+        const jwtUser = await this.usersService.findByUsername2JWTUser(
+          jobParams.username as string,
+        );
+        jobUser = jwtUser;
+      } else if (requestUser) {
+        jobUser = requestUser;
+      }
+      if (jobUser) {
+        newBody = {
+          ...newBody,
+          ownerUser: jobUser?.username,
+        };
+      }
+      // ensure compatibility with v4 which requires ownerGroup in the dto of jobs created by normal user
+      if (Object.keys(jobParams).includes("ownerGroup")) {
+        newBody = {
+          ...newBody,
+          ownerGroup: jobParams.ownerGroup as string,
+        };
+      } else if (Array.isArray(jobParams.datasetList)) {
+        if (jobConfig.create.auth === "#datasetAccess") {
+          const datasetGroups = [];
+          for (const datasetDto of jobParams.datasetList) {
+            if (datasetDto.pid) {
+              const dataset = await this.datasetsService.findOne({
+                where: { pid: datasetDto.pid },
+              });
+              datasetGroups.push([
+                ...(dataset?.accessGroups ?? []),
+                dataset?.ownerGroup,
+              ]);
+            }
+          }
+          const commonDatasetGroups = intersection(datasetGroups);
+          const commonGroups = intersection([
+            commonDatasetGroups,
+            jobUser?.currentGroups ?? [],
+          ]);
```

Comment:
This can be simplified into a single call
```suggestion
          const commonGroups = intersection([
            ...datasetGroups,
            jobUser?.currentGroups ?? [],
          ]);
```